variables.tf

```diff
@@ -1,3 +1,15 @@
+variable "availability_zones" {
+  type        = list(string)
+  default     = []
```

Comment:
Maybe default to null, which means you did not set the value. Seems like a good default since its an optional field.